### PR TITLE
Java & Ruby Documentation

### PR DESCRIPTION
**Pull request by Thiyagu55**
Converted Hugo format docs from https://google.github.io/sqlcommenter/ into markdown

### CODE REVIEW

File: docs/java/spring/README.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        registry.addInterceptor(sqlInterceptor());
+    }
+}
+```
+
+### Before Spring 5
+
+If using a version before Spring 5, your `WebConfig` class needs to extend the [WebMVCConfigureAdapter](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/servlet/config/annotation/WebMvcConfigurerAdapter.html) class instead like this:
+
+``` java
```
Comment (sjs994) [inline docs/java/spring/README.md:99]: nit: any reason to have a space in between ?
to maintain consistency with other language code block syntax, maybe  the space should be removed.

File: docs/java/spring/README.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    }
+}
+```
+
+## XML based configuration
+
+You can add the interceptor as a bean in your XML configuration
+
+### For_Every_Method
+``` xml
```
Comment (sjs994) [inline docs/java/spring/README.md:123]: nit: Same as above

File: docs/java/hibernate/README.md [hunk 3]
```diff
@@ -0,0 +1,1143 @@
...
+        return new String[]{
+            "com.google.cloud.sqlcommenter.spring.jpa.domain"
+        };
+    }
+}
+````
+
+Now, the test looks as follows:
+
+````java
```
Comment (sjs994) [inline docs/java/hibernate/README.md:393]: there seems to be an extra `

File: docs/java/hibernate/README.md [hunk 4]
```diff
@@ -0,0 +1,1143 @@
...
+    @Transactional
+    public Post findPostById(Long id) {
+        return postRepository.findById(id).orElse(null);
+    }
+}
+```
+
+The Spring JPA configuration looks as follows:
+
+````java
```
Comment (sjs994) [inline docs/java/hibernate/README.md:276]: \```java

File: docs/java/hibernate/README.md [hunk 5]
```diff
@@ -0,0 +1,1143 @@
...
+                    )
+                    .count()
+        );
+    }
+}
+````
+
+When running the unit test above, we can see that the SQL statements include the comments as well:
+
+````sql
```
Comment (sjs994) [inline docs/java/hibernate/README.md:527]: \```sql

**Review by sjs994 — COMMENTED** [review]
Thanks @Thiyagu55. Looks good to me.
Few minor comments to make the code blocks syntax consistent.

**Review by sjs994 — APPROVED** [review]

**Review by sjs994 — APPROVED** [review]

File: docs/ruby/README.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+[![](../images/activerecord_marginalia-logo.png)](ruby-on-rails)
+
+<style>
+    img[src*='/activerecord_marginalia-logo.png']{
+        max-width: 30%;
+        float: None;
+        margin:0 2%;
+    }
+
+</style>
```
Comment (lonewolf3739) [inline docs/ruby/README.md:15]: Github strips these tags and also there is no image `activerecord_marginalia-logo.png `

**Review by lonewolf3739 — APPROVED** [review]

File: docs/ruby/README.md (hunk 6 above)
Comment (sjs994) [reply docs/ruby/README.md:15]: @lonewolf3739 the image is there in the location: `docs/images/activerecord_marginalia-logo.png` in the commit section.

Also for reference: this is view when the doc is deployed with gh-pages: [sqlcommenter-doc-preview](https://thiyagu55.github.io/opentelemetry-sqlcommenter/ruby/)

**Pull request merged**